modules/settings/crew-settings.js

```diff
@@ -0,0 +1,94 @@
+export default class CrewSettings extends FormApplication {
+  /** @override */
+  static get defaultOptions() {
+    return mergeObject(super.defaultOptions, {
+      id: "data-importer",
+      classes: ["starwarsffg", "data-import"],
+      title: `${game.i18n.localize("SWFFG.UISettingsLabel")}`,
+      height: 265,
+      resizable: true,
+      template: "systems/starwarsffg/templates/dialogs/crew-settings.html"
+    });
+  }
+
+  getData(options) {
+    const gs = game.settings;
+    const canConfigure = game.user.can("SETTINGS_MODIFY");
+
+    const data = {
+      system: { title: game.system.title, menus: [], settings: [] },
+    };
+
+    // Classify all settings
+    for (let setting of gs.settings.values()) {
+      // Exclude settings the user cannot change
+      if (!setting.key.includes("arrayCrewRoles") || (!canConfigure && setting.scope !== "client")) continue;
+
+      // Update setting data
+      const s = duplicate(setting);
+      s.name = game.i18n.localize(s.name);
+      s.hint = game.i18n.localize(s.hint);
+      s.value = game.settings.get(s.module, s.key);
+      s.type = setting.type instanceof Function ? setting.type.name : "String";
+      s.isCheckbox = setting.type === Boolean;
+      s.isSelect = s.choices !== undefined;
+      s.isRange = setting.type === Number && s.range;
+      s.isFilePicker = setting.valueType === "FilePicker";
+
+      // Classify setting
+      const name = s.module;
+      if (name === game.system.id && s.key.includes("arrayCrewRoles")) data.system.settings.push(s);
+    }
```

Comment:
Not sure I understand what's going on here

Reply:
it's copy/pasted from another settings file :p

Reply:
As far as I can tell, it seems like a very complicated way of saying `data.system.settings = game.settings.get('starwarsffg', 'arrayCrewRoles');` 😕